tools/fv_io.F90

```diff
@@ -590,34 +712,117 @@ subroutine  fv_io_write_restart(Atm, timestamp)
 
     type(fv_atmos_type),        intent(inout) :: Atm
     character(len=*), optional, intent(in) :: timestamp
+    integer :: ntiles
+    logical :: tile_file_exists
+    type(domain2d)     :: fv_domain
+    character(len=120) :: fname
+    character(len=20) :: suffix
+    character(len=1) :: tile_num
+    fv_domain = Atm%domain
 
 !!$    if ( use_ncep_sst .or. Atm%flagstruct%nudge .or. Atm%flagstruct%ncep_ic ) then
 !!$       call mpp_error(NOTE, 'READING FROM SST_RESTART DISABLED')
 !!$       !call save_restart(Atm%SST_restart, timestamp)
 !!$    endif
 
     if ( (use_ncep_sst .or. Atm%flagstruct%nudge) .and. .not. Atm%gridstruct%nested ) then
-       call save_restart(Atm%SST_restart, timestamp)
+       !call save_restart(Atm%SST_restart, timestamp)
     endif
 
-    call save_restart(Atm%Fv_restart, timestamp)
-    call save_restart(Atm%Fv_tile_restart, timestamp)
-    call save_restart(Atm%Rsf_restart, timestamp)
+    suffix = ''
+    if (present(timestamp)) then
+      fname = 'RESTART/'//trim(timestamp)//'.fv_core.res'//trim(suffix)//'.nc'
+    else
+      fname = 'RESTART/fv_core.res'//trim(suffix)//'.nc'
+    endif
+    Atm%Fv_restart_is_open = open_file(Atm%Fv_restart, fname, "overwrite", is_restart=.true.)
```

Comment:
With this every rank is going to write the restart, so add the pelist argument to the open_file call.